messagequeue/messagequeue.go

```diff
@@ -156,6 +163,9 @@ func (mq *MessageQueue) runQueue() {
 				for {
 					_, metadata, err := mq.extractOutgoingMessage()
 					if err == nil {
+						span := trace.SpanFromContext(metadata.ctx)
+						span.SetStatus(codes.Error, "message queue shutdown")
```

Comment:
You could also `RecordError()` with the error made 2 lines down for `publishError()`
I don't know what consuming tools do with these but there are separate fields where errors are recorded in the data, aside from just status codes